src/summarize.jl

```diff
@@ -134,13 +134,13 @@ Summarize `chains` in a `ChainsDataFrame`.
 * `summarize(chns; sections=[:parameters, :internals])` : Chain summary for multiple sections
 """
 function summarize(
-    chains::Chains, funs...;
+    chains::Chains{T,A,L,K,I,W}, funs::Base.Callable...;
```

Comment:
We don't want to restrict the function to `Base.Callable`. Probably it would also be better to dispatch on the weights in the function body when needed as large parts are the same, regardless of the weights. Maybe not even necessary to dispatch on the weights at all.
```suggestion
    chains::Chains{T,A,L,K,I,W}, funs...;
```